nemo_skills/pipeline/eval.py

```diff
@@ -409,46 +410,160 @@ def eval(
             assert benchmark_args.eval_subfolder.startswith("tmp-")
             benchmark_args.eval_subfolder = benchmark_args.eval_subfolder[4:]
             judge_pipeline_args["output_dir"] = str(Path(output_dir) / benchmark_args.eval_subfolder)
-            judge_ctx = deepcopy(ctx)
-            # removing any extra arguments here as they are assumed to be for the main job
-            judge_ctx.args = []
-            if judge_wrap_args:
-                judge_ctx.args.extend(judge_wrap_args.split(" "))
-            if extra_judge_args:
-                judge_ctx.args.extend(extra_judge_args.split(" "))
-
-            # the default parameters always have server_address, but it needs to be removed if model is self-hosted
-            if judge_server_gpus is not None:
-                judge_pipeline_args["server_address"] = None
-
-            for judge_server_param, judge_server_value in cli_judge_pipeline_args.items():
-                if judge_server_value is not None:
-                    judge_pipeline_args[judge_server_param] = judge_server_value
-            # TODO: should we support parsing a string?
-            if extra_judge_pipeline_args is not None:
-                judge_pipeline_args.update(extra_judge_pipeline_args)
-            has_tasks = True
-            judge_tasks = _generate(
-                ctx=judge_ctx,
-                expname=f"{expname}-{benchmark}-judge",
-                log_dir=log_dir + "/judge",
-                cluster=cluster,
-                config_dir=config_dir,
-                partition=partition,
-                with_sandbox=with_sandbox,
-                keep_mounts_for_sandbox=keep_mounts_for_sandbox,
-                run_after=run_after,
-                reuse_code_exp=reuse_code_exp,
-                reuse_code=reuse_code,
-                exclusive=exclusive,
-                installation_command=installation_command,
-                sbatch_kwargs=sbatch_kwargs,
-                _reuse_exp=exp,
-                _task_dependencies=(
-                    dependent_tasks if cluster_config["executor"] == "slurm" else all_tasks + _task_dependencies
-                ),
-                **judge_pipeline_args,
-            )
+
+            # Check for per-benchmark judge_type, fall back to global judge_type
+            benchmark_judge_type = judge_pipeline_args.pop("judge_type", judge_type)
+
+            # creating custom command for nvembed judge type
+            if benchmark_judge_type == "nvembed":
```

Comment:
could you please try to refactor this into a function as we'd have more judge types in the future and it will be convenient if all this is handled in a separate function